src/document.cpp

```diff
@@ -368,6 +376,16 @@ rvoe<NoReturnValue> PdfDocument::add_page(std::string resource_dict,
     RETOK;
 }
 
+rvoe<NoReturnValue>
+PdfDocument::add_page_labeling(uint32_t start_page,
+                               std::optional<CapyPDF_PageLabelNumberStyle> style,
+                               std::optional<u8string> prefix,
+                               std::optional<uint32_t> start_num) {
+    page_labels.emplace_back(start_page, style, prefix, start_num);
```

Comment:
This should check if `page_labels` is empty and if it is not, it should check that `page_labels.back() .start_page < start_page`. AFAICT the values in a number tree must be strictly increasing.